publish-mqtt/src/homie.rs

```diff
@@ -0,0 +1,204 @@
+use async_channel::{SendError, Sender};
+use rumqttc::{self, EventLoop, LastWill, MqttOptions, Publish, QoS, Request};
+use std::error::Error;
+use tokio::task::{self, JoinHandle};
+
+const HOMIE_VERSION: &str = "4.0";
+
+#[derive(Clone, Debug)]
+pub struct Property {
+    id: String,
+    name: String,
+    datatype: String,
+    unit: String,
+}
+
+impl Property {
+    pub fn new(id: &str, name: &str, datatype: &str, unit: &str) -> Property {
+        Property {
+            id: id.to_string(),
+            name: name.to_string(),
+            datatype: datatype.to_string(),
+            unit: unit.to_string(),
```

Comment:
(optional) if you don't like writing `.to_string()` everywhere, you can use `.to_owned()` which does the same thing in this case, but is semantically closer to what you're doing.

Reply:
Done.